src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -719,11 +703,13 @@ public override Vertex WriteVertex(NodeFactory factory)
 
             VertexBag layoutInfo = new VertexBag();
 
-            KeyValuePair<int, NativeLayoutVertexNode>[] dictionaryEntries = new KeyValuePair<int, NativeLayoutVertexNode>[_genericDictionaryEntries.Count];
+            DictionaryLayoutNode associatedLayout = factory.GenericDictionaryLayout(_method);
+            KeyValuePair<int, NativeLayoutVertexNode>[] dictionaryEntries = new KeyValuePair<int, NativeLayoutVertexNode>[associatedLayout.Entries.Count];
             int index = 0;
-            foreach (KeyValuePair<int, NativeLayoutVertexNode> entry in _genericDictionaryEntries)
+            foreach (GenericLookupResult entry in associatedLayout.Entries)
```

Comment:
We will need some sort of extensibility point... I think it will need to have the dictionarylayout able to express the template NativeLayotVertexNodes itself.